frontend/src/modules/entitydetails/components/EntityDetails.js

```diff
@@ -97,6 +97,20 @@ export class EntityDetailsBase extends React.Component<InternalProps, State> {
         dispatch(otherlocales.actions.get(parameters.entity, parameters.locale));
     }
 
+    customMachinerySearch = (query: string) => {
+        const { dispatch, locale, selectedEntity } = this.props;
+
+        // Deep copy entity to avoid mutating state
+        const entity = JSON.parse(JSON.stringify(selectedEntity));
```

Comment:
A more elegant way to do this with ES6 is to use the spread operator: `const entity = { ...selectedEntity };`